processing/src/test/java/org/apache/druid/java/util/metrics/HttpPostEmitterMonitorTest.java

```diff
@@ -0,0 +1,100 @@
+    when(mockHttpPostEmitter.getTotalDroppedBuffers()).thenReturn(10);
+    when(mockHttpPostEmitter.getTotalAllocatedBuffers()).thenReturn(20);
+    when(mockHttpPostEmitter.getTotalFailedBuffers()).thenReturn(5);
+    when(mockHttpPostEmitter.getBatchFillingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getSuccessfulSendingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getFailedSendingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getEventsToEmit()).thenReturn(200L);
+    when(mockHttpPostEmitter.getLargeEventsToEmit()).thenReturn(75L);
+    when(mockHttpPostEmitter.getBuffersToEmit()).thenReturn(30);
+    when(mockHttpPostEmitter.getBuffersToReuse()).thenReturn(15);
+
+    final StubServiceEmitter stubServiceEmitter = new StubServiceEmitter("service", "host");
+
+    assertTrue(monitor.doMonitor(stubServiceEmitter));
+
+    Map<String, List<StubServiceEmitter.ServiceMetricEventSnapshot>> metricEvents = stubServiceEmitter.getMetricEvents();
+
+    assertMetricNameAndValue(metricEvents, "emitter/successfulSending/maxTimeMs", 0);
+    assertMetricNameAndValue(metricEvents, "emitter/events/emitted/delta", 100L);
+    assertMetricNameAndValue(metricEvents, "emitter/successfulSending/minTimeMs", 0);
+    assertMetricNameAndValue(metricEvents, "emitter/buffers/emitQueue", 30);
+    assertMetricNameAndValue(metricEvents, "emitter/failedSending/minTimeMs", 0);
+    assertMetricNameAndValue(metricEvents, "emitter/buffers/allocated/delta", 20);
+    assertMetricNameAndValue(metricEvents, "emitter/batchFilling/maxTimeMs", 0);
+    assertMetricNameAndValue(metricEvents, "emitter/buffers/dropped/delta", 10);
+    assertMetricNameAndValue(metricEvents, "emitter/batchFilling/minTimeMs", 0);
+    assertMetricNameAndValue(metricEvents, "emitter/events/emitQueue", 200L);
+    assertMetricNameAndValue(metricEvents, "emitter/events/large/emitQueue", 75L);
+    assertMetricNameAndValue(metricEvents, "emitter/buffers/reuseQueue", 15);
+    assertMetricNameAndValue(metricEvents, "emitter/buffers/failed/delta", 5);
+    assertMetricNameAndValue(metricEvents, "emitter/failedSending/maxTimeMs", 0L);
+  }
+
+  private void assertMetricNameAndValue(Map<String, List<StubServiceEmitter.ServiceMetricEventSnapshot>> metricEvents, String metricName, Number expectedValue)
+  {
+    assertEquals(metricEvents.get(metricName).get(0).getMetricEvent().getMetric(), metricName);
+
+    if (metricEvents.get(metricName).get(0).getMetricEvent().getValue() instanceof Long)
+    {
+      assertEquals(metricEvents.get(metricName).get(0).getMetricEvent().getValue(), expectedValue.longValue());
+    }
+    else if (metricEvents.get(metricName).get(0).getMetricEvent().getValue() instanceof Integer)
+    {
+      assertEquals(metricEvents.get(metricName).get(0).getMetricEvent().getValue(), expectedValue.intValue());
+    }
```

Comment:
I think these can be collapsed into the following:
```java
assertEquals(metricEvents.get(metricName).get(0).getMetricEvent().getValue().doubleValue(), expectedValue.doubleValue());
```

Can you try this?

Reply:
Yup, this would work. Pushed in the next commit.